feat(TreeViewLeaf): show a delete button in active state

Lelith opened the pull request:
**Who**: @lelith

**What / Why**: 
- shows a delete button when node is set as active
- the delete button accepts onDelete click-function
- contains also design adjustment the style on the NumberPicker Buttons.

![vollbild_12_06_18__10_25](https://user-images.githubusercontent.com/1789174/41278771-fbc8037e-6e2a-11e8-8c05-774708d35a57.png)

eleventhirtyfour commented:
@Lelith  the small delete icon on the top right is ment to delete the whole node?

Lelith commented:
@eleventhirtyfour exactly, but the delete will also appear on the nodes in the lower tree levels, so you can delete every level individually if needed.
![storybook](https://user-images.githubusercontent.com/1789174/41235755-b9b2cb04-6d8f-11e8-9c18-5f949b4d465f.png)

eleventhirtyfour commented:
@Lelith  ok but for now we are not considering the design and the ux right? cause so smaller than the number picker and randomly floating around looks a bit bad

Lelith commented:
@eleventhirtyfour i m open for suggestions about the position of the cross element.

cfleschhut commented on lib/components/TreeView/TreeView.scss:
```diff
@@ -157,6 +163,45 @@
+    top: 0;
+    right: 0;
+    width: 15px;
+    height: 15px;
```
@Lelith Verrry fine detail here again, but can we decrease the width & height to `14px`?

With even instead of odd numbers for the sizing the delete icon (`x`) will appear more evenly centered:

<img width="776" alt="bildschirmfoto 2018-06-11 um 18 12 01" src="https://user-images.githubusercontent.com/34942/41243684-11212792-6da3-11e8-9067-85252a8cf971.png">

Lelith commented:
@cfleschhut:  I adjusted the sizing a bit to make it all with even numbers, also adjusted the number picker buttons to be even numbers and changed to font-family to one that has a more square-shape on the plus/minus symbols. Thank you for your suggestions. 

if the button works as tought we can only see after the user testing and then move it to another place or change it completely.

Lelith commented:
:tada: This PR is included in version 1.2.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/gsy-component-library)
- [GitHub release](https://github.com/gridsingularity/component-library/releases/tag/v1.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
